src/testing/cluster/storage_checker.zig

```diff
@@ -210,6 +210,8 @@ pub fn StorageCheckerType(comptime Storage: type) type {
                 assert(block_header.op == block_address);
 
                 checksum ^= vsr.checksum(block[0..block_header.size]);
+                // Extra guard against identical blocks:
+                checksum ^= vsr.checksum(std.mem.asBytes(&block_address));
```

Comment:
I think it's not quite it, this will give identical checksums:

R0: addr=1, content=X; addr=2, content=X
R1: addr=1, content=Y; addr=2, content=Y

basically, you are independency checksumming block contents and block addresses, but we want checksum them entangled, so that indentical contents doesn't cancel out. We can do it like this:

```zig
checksum ^= vsr.checksum(block[0..block_header.size] ++ std.mem.asBytes(&block_address))
```

but, looking at the bigger context, perhaps we need a streaming variant of our checksum?

```zig
pub fn checksum(source: []const u8) u128 {
    var stream = ChecksumStream.init();
    stream.add(source);
    stream.checksum();
}

pub const ChecksumStream = struct {
    state: Aegis128LMac_128,

    pub fn init() ChecksumStream {
        seed_init();
        return ChecksumStream{ .state = seed_state };
    }

    pub fn add(stream: *ChecksumStream, bytes: []const u8) void {
        stream.state.update(bytes);
    }

    pub fn checksum(stream: *ChecksumStream) u128 {
        var result: u128 = undefined;
        stream.state.final(mem.asBytes(&result));
        stream.* = undefined;
        return result;
    }
};
```